### PR TITLE
Make sure that we don't pipe stdin when starting a job

### PR DESCRIPTION
**Pull request by ddeville**
Starting with version 13, ripgrep always stats stdin and if it's not a TTY it uses it to read data. Unfortunately, Neovim always attaches a pipe to stdin by default and that leads to ripgrep reading nothing and it essentially breaks vim-grepper when targeting a recent version of ripgrep.

(see https://github.com/BurntSushi/ripgrep/issues/1892 for more info)

This was fixed in nvim by adding an option to jobstart to not pipe stdin (see https://github.com/neovim/neovim/pull/14812). So we use this here which I verified fixes search through rg.

Note that I'm not 100% sure how to gate this. It was technically added as a commit to neovim after 0.5 shipped so I imagine it will technically be released in 0.5.1. But it should also be harmless to only gate this to `nvim` since the options are a dictionary and old versions of neovim will just ignore that `stdin` key.

This fixes https://github.com/mhinz/vim-grepper/issues/244

### CODE REVIEW

File: plugin/grepper.vim
Comment (blueyed) [inline plugin/grepper.vim:943]:
```suggestion
      " (see https://github.com/mhinz/vim-grepper/issues/244 for more info)
```
The PR is referenced twice - maybe you've meant the issue here?  Or the one in ripgrep?

**Review by blueyed — APPROVED** [review]
Confirmed to work with updated Neovim master.

Thanks!

File: plugin/grepper.vim
Comment (ddeville) [reply plugin/grepper.vim]: Thanks for catching this, just updated the PR!

**Review by blueyed — APPROVED** [review]

**Comment by blueyed** [comment]
Great, thanks again.
Let's wait for @mhinz to approve/merge it (I cannot).

**Comment by lbonn** [comment]
I have merge rights, just one question, will `if has('nvim-0.5.1')` also work with later versions?

**Comment by ddeville** [comment]
Yes it will. I’m running the nightly 0.6 version and it works fine.

**Pull request merged**
